pydis_site/apps/events/views/page.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+
+from django.conf import settings
+from django.core.handlers.wsgi import WSGIRequest
+from django.http import Http404, HttpResponse
+from django.template import Context, Template
+from django.views import View
+
+PAGES_PATH = Path(settings.BASE_DIR, "pydis_site", "apps", "events", "pages")
+
+
+class PageView(View):
+    """Handles event pages showing."""
+
+    def get(self, request: WSGIRequest, path: str) -> HttpResponse:
+        """Render event page rendering based on path."""
+        # We need to get rid from trailing slash when path have this
+        if path.endswith("/"):
+            path = path[:-1]
+
+        page_path = PAGES_PATH.joinpath(path)
```

Comment:
You can use `pathlib.Path`'s `__div__` overrides here, it's generally a bit cleaner than using `joinpath`.
```suggestion
        page_path = PAGES_PATH / path
```